ui/src/components/EdgePinCluster.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from "react";
+import L, { LatLng } from "leaflet";
+import "react-leaflet-markercluster/dist/styles.min.css";
+import "leaflet.markercluster";
+import MarkerClusterGroup from "react-leaflet-markercluster";
+import "leaflet/dist/leaflet.css";
+import { EdgePin, getAnchor } from "./EdgePin";
+import { EdgeType } from "./EdgePins";
+import { arrow } from "./Arrow";
+
+const createClusterCustomIcon = function (edgeType: EdgeType) {
+  let icon;
+  switch (edgeType) {
+    case EdgeType.Right:
+      icon = arrow(0);
+      break;
+    case EdgeType.Bottom:
+      icon = arrow(90);
+      break;
+    case EdgeType.Top:
+      icon = arrow(270);
+      break;
+    case EdgeType.Left:
+      icon = arrow(180);
+      break;
+  }
+
+  return new L.Icon({
+    iconUrl: icon,
+    iconRetinaUrl: icon,
+    iconAnchor: getAnchor(edgeType),
+    popupAnchor: [-234, 200],
+    shadowUrl: null,
+    shadowSize: null,
+    shadowAnchor: null,
+    iconSize: new L.Point(60, 70),
+  });
+};
+
+export interface EdgePinClusterProps {
+  clusterData: { pos: LatLng; angle: number; edgeType: EdgeType }[];
+}
+
+export function EdgePinCluster({
+  clusterData,
+}: EdgePinClusterProps): JSX.Element {
+  const rightCluster = clusterData.filter(
+    (data) => data.edgeType === EdgeType.Right
+  );
+  const leftCluster = clusterData.filter(
+    (data) => data.edgeType === EdgeType.Left
+  );
+  const topCluster = clusterData.filter(
+    (data) => data.edgeType === EdgeType.Top
+  );
+  const bottomCluster = clusterData.filter(
+    (data) => data.edgeType === EdgeType.Bottom
+  );
+  return (
+    <React.Fragment>
+      <MarkerClusterGroup
+        showCoverageOnHover={false}
+        spiderLegPolylineOptions={{ opacity: 0 }}
+        iconCreateFunction={() => createClusterCustomIcon(EdgeType.Right)}
+      >
```

Comment:
Hmm can we just do this in a loop if everything stays the same but `EdgeType` and the array thats being mapped over?

Something like (sorry for bad formatting lol):

```ts
const arr = [{type: EdgeType.Right, cluster: rightCluster}, {type: EdgeType.Left, cluster: leftCluster} ... etc]
arr.map(({ type, cluster }) => {
  <MarkerClusterGroup
        showCoverageOnHover={false}
        spiderLegPolylineOptions={{ opacity: 0 }}
        iconCreateFunction={() => createClusterCustomIcon(type)}
      >
  {cluster.map((marker, key) => {
          return (
            <EdgePin
              key={key}
              position={marker.pos}
              angle={marker.angle}
              edgeType={marker.edgeType}
            />
          );
        })}
   </MarkerClusterGroup>
});
```